Unix/protocol/protocol.c

```diff
@@ -912,13 +920,31 @@ static MI_Boolean _ProcessAuthMessage(
     /* server waiting client's first request? */
     if (PRT_AUTH_WAIT_CONNECTION_REQUEST == handler->clientAuthState)
     {
-        return _ProcessAuthMessageWaitingConnectRequest(handler, binMsg);
+        MI_Boolean keepConnection;
+        MI_Boolean r;
+
+        r = _ProcessAuthMessageWaitingConnectRequest(handler, binMsg, &keepConnection);
+        if (r && !keepConnection)
+        {
+            handler->strand.info.otherClosedThis = MI_TRUE;
```

Comment:
unfortunately you should not exit these directly. This implies you need to call Strand_Close somewhere